### PR TITLE
Make SVI/Income/Buildings optional when possible

### PR DESCRIPTION
**Pull request by janakdr**
SVI and Income are always optional, Buildings optional when damage asset not present. Messaging stays in the button. We change the button color to a grayish yellow when missing optional assets.

### CODE REVIEW

File: docs/import/manage_disaster.js [hunk 1]
```diff
@@ -61,48 +61,90 @@ let scoreBoundsMap;
...
- * all validation succeeds, enables kick-off button, otherwise disables and
- * changes button text to say what is missing.
+ * Checks that all mandatory fields that can be entered by the user have
+ * non-empty values. Does not check that assets actually exist, are of valid
+ * type, etc. If all validation succeeds, enables kick-off button, otherwise
+ * disables and changes button text to say what is missing.
+ *
+ * Some fields (Income, SVI) are optional. If they are missing, a separate
+ * message is displayed on the button, but it can still be enabled. If it is
+ * enabled, it is yellowed a bit to indicate the missing optional assets.
```
Comment (juliexxia) [inline docs/import/manage_disaster.js]: So the color of the button for unable to kick off and able but missing is the same? What would you think about making them different? I think my slight preference is for anytime the button can actually be used for kicking off, it's green. But @ruthtalbot what do you think?

File: docs/import/manage_disaster.js [hunk 2]
```diff
@@ -61,48 +61,90 @@ let scoreBoundsMap;
...
+ * The buildings asset is optional if the damage asset is not present, but is
+ * required if the damage asset is present.
  */
 function validateUserFields() {
   const states = disasterData.get(getDisaster()).states;
   const missingItems = [];
-  for (const scoreAssetType of scoreAssetTypes) {
+  const damageAssetPresent = !!$('#damage-asset-select').val();
```
Comment (juliexxia) [inline docs/import/manage_disaster.js]: nit: would you mind also doing  this in update.js for the initialization of the hasDamageAset var?

Comment (juliexxia) [inline docs/import/manage_disaster.js]: can you document what the missingItems array ends up holding before we start processing it?

File: docs/import/manage_disaster.js [hunk 3]
```diff
@@ -61,48 +61,90 @@ let scoreBoundsMap;
...
-  const hasDamage = $('#damage-asset-select').val() ||
-      getElementFromPath(scoreCoordinatesPath);
+  const hasDamage =
+      damageAssetPresent || getElementFromPath(scoreCoordinatesPath);
   let message = '';
+  let optionalMessage = '';
   if (missingItems.length) {
-    message = 'Missing asset(s): ' + missingItems.join(', ');
+    for (const missingItem of missingItems) {
+      const optionalBuildings =
```
Comment (juliexxia) [inline docs/import/manage_disaster.js]: This logic is starting to get quite hairy (though looks good in the button!) If we have time (lol) maybe we can think about more visual messaging through icons or something.

File: cypress/integration/unit_tests/manage_disaster_set_score_parameters_test.js [hunk 4]
```diff
@@ -69,13 +69,18 @@ describe('Score parameters-related tests for manage_disaster.js', () => {
...
+  const alwaysOptionalMissing =
+      '; warning: created asset will be missing Income, SVI';
+  const allOptionalMissing = alwaysOptionalMissing + ', Building counts';
+  const allStateAssetsMissingWithDamageAssetText = allMandatoryMissingText +
+      ', Microsoft Building Shapefiles' + alwaysOptionalMissing;
+  const allMissingText = allMandatoryMissingText +
+      ', and must specify either damage asset or map bounds' +
+      allOptionalMissing;
+
+  it.only('validates asset data', () => {
```
Comment (juliexxia) [inline cypress/integration/unit_tests/manage_disaster_set_score_parameters_test.js]: rm only

File: docs/import/manage_disaster.js (hunk 1 above)
Comment (janakdr) [reply docs/import/manage_disaster.js]: It's supposed to be different: gray vs yellow vs green. There was a bug where if it was yellow, when it was disabled it stayed yellow, so thanks for pointing that out! Added an assertion in the test for that case.

@ruthtalbot seemed ok with yellow as opposed to just green (that's my preference as well) in the chat.

File: docs/import/manage_disaster.js (hunk 2 above)
Comment (janakdr) [reply docs/import/manage_disaster.js]: Prefer to not touch that file in this PR, that ok?

Comment (janakdr) [reply docs/import/manage_disaster.js]: Added a comment at the top.

File: docs/import/manage_disaster.js (hunk 3 above)
Comment (janakdr) [reply docs/import/manage_disaster.js]: Agree it's getting hairy, wish I had an easier way. Can think about a refactoring soon.

File: docs/import/manage_disaster.js (hunk 1 above)
Comment (juliexxia) [reply docs/import/manage_disaster.js]: Oh yeah, that sounds great too, as long as there's a difference between kicking off with all assets and kicking off with some optionals missing.

File: docs/import/manage_disaster.js (hunk 2 above)
Comment (janakdr) [reply docs/import/manage_disaster.js]: https://github.com/givedirectly/Google-Partnership/pull/388

**Pull request merged**